core/optaplanner-core-impl/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -77,7 +76,7 @@ public class ScoreDirectorFactoryConfig extends AbstractConfig<ScoreDirectorFact
     @XmlElement(name = "scoreDrlFile")
     protected List<File> scoreDrlFileList = null;
     @XmlTransient
-    protected Supplier<KieBase> gizmoKieBaseSupplier = null;
```

Comment:
Technically this is backwards incompatible. But this API is clearly marked as internal by a comment, and it is also the only way to make the kie-api dependency optional. Therefore, I am willing to live with this change.

Reply:
I agree.